src/core/tests/Test_StatusHandlerTruncation.py

```diff
@@ -0,0 +1,784 @@
+            truncated_substatus_file_data = json.load(file_handle)[0]
+
+        self.__assert_patch_summary_from_status(truncated_substatus_file_data, Constants.INSTALLATION, Constants.PATCH_INSTALLATION_SUMMARY, Constants.STATUS_ERROR, patch_count,
+            errors_count=5, errors_code=Constants.PatchOperationTopLevelErrorCode.ERROR, complete_substatus_file_data=complete_substatus_file_data, is_under_internal_size_limit=True, is_truncated=True)
+
+        # Assert all installation fields in the message json are equal in both status files
+        self.__assert_installation_truncated_msg_fields(complete_substatus_file_data, truncated_substatus_file_data)
+
+        # Assert 'Count of patches removed from: [Assessment=0] [Installation=xxx] log message is called
+        self.__assert_truncated_patches_removed(truncated_substatus_file_data, patch_count_installation=patch_count)
+
+        # Reset sys.stdout, close and delete tmp file
+        # Note: sys log/print will not display in terminal, for debugging comment __remove_tmp_file_reset_stdout and check the tmp file
+        self.__remove_tmp_file_reset_stdout()
+
+    def test_both_assessment_and_installation_over_size_limit_truncated(self):
+        """ Perform truncation on assessment/installation patches.
+        Before truncation: 700 assessment patches in status, 1200 installation patches in status
+        complete status file byte size: 242kb,
+        Expected (After truncation): 374 assessment patches in status, 250 installation patches in status
+        operation: Installation,
+        substatus name: [assessment=PatchAssessmentSummary][installation=PatchInstallationSummary],
+        substatus status: [assessment=warning][installation=warning],
+        errors code: [assessment=0 (success)][installation=0 (success)],
+        errors details count: [assessment=0][installation=0],
+        count of patches removed from log: [assessment=326[installation=950],
+        truncated status file byte size: 126kb. """
+
+        self.runtime.execution_config.operation = Constants.INSTALLATION
+        self.runtime.status_handler.set_current_operation(Constants.INSTALLATION)
+        # Set up tmp file store all log/print message and set sys.stdout point to it.
+        # Note: sys log/print will not display in terminal, for debugging comment __remove_tmp_file_reset_stdout and check the tmp file
+        self.__create_tmp_file_for_log_and_set_stdout()
+
+        patch_count_assessment = 700
+        test_packages, test_package_versions = self.__set_up_packages_func(patch_count_assessment)
+        self.runtime.status_handler.set_package_assessment_status(test_packages, test_package_versions)
+        self.runtime.status_handler.set_assessment_substatus_json(status=Constants.STATUS_SUCCESS)
+
+        patch_count_pending = 250
+        patch_count_installed = 250
+        patch_count_installation = patch_count_pending + patch_count_installed
```

Comment:
Function comment mentions 1200 installation patches before truncation. Update the comment to reflect correct values

Reply:
done